core/src/components/picker-column-internal/picker-column-internal.tsx

```diff
@@ -0,0 +1,249 @@
+      // (Vertical offset from parent) - (three empty picker rows) + (half the height of the target to ensure the scroll triggers)
+      top: target.offsetTop - (3 * target.clientHeight) + (target.clientHeight / 2),
+      left: 0,
+      behavior: smooth ? 'smooth' : undefined
+    });
+  }
+
+  /**
+   * When ionInputModeChange is emitted, each column
+   * needs to check if it is the one being made available
+   * for text entry.
+   */
+  private inputModeChange = (ev: PickerInternalCustomEvent) => {
+    if (!this.numericInput) { return; }
+
+    const { inputMode, inputModeColumn } = ev.detail;
+
+    /**
+     * If inputModeColumn is undefined then this means
+     * all numericInput columns are being selected.
+     */
+    const isColumnActive = inputModeColumn === undefined || inputModeColumn === this.el;
+
+    if (!inputMode || !isColumnActive) {
+      this.isActive = false;
+      return;
+    }
+
+    this.isActive = true;
+  }
+
+  /**
+   * When the column scrolls, the component
+   * needs to determine which item is centered
+   * in the view and will emit an ionChange with
+   * the item object.
+   */
+  private initializeScrollListener = () => {
+    const { el } = this;
+
+    let timeout: any;
+    let activeEl: HTMLElement | null = this.activeItem;
+    const scrollCallback = () => {
+      raf(() => {
+        if (timeout) {
+          clearTimeout(timeout);
+          timeout = undefined;
+        }
+
+        /**
+         * Select item in the center of the column
+         * which is the month/year that we want to select
+         */
+        const bbox = el.getBoundingClientRect();
+        const centerX = bbox.x + (bbox.width / 2);
+        const centerY = bbox.y + (bbox.height / 2);
+
+        const activeElement = el.shadowRoot!.elementFromPoint(centerX, centerY) as HTMLElement;
+        if (activeEl !== null) {
+          activeEl.classList.remove(PICKER_COL_ACTIVE);
+        }
+
+        activeEl = activeElement;
+        activeElement.classList.add(PICKER_COL_ACTIVE);
+
+        timeout = setTimeout(() => {
+          const dataIndex = activeElement.getAttribute('data-index');
+
+          /**
+           * If no value it is
+           * possible we hit one of the
+           * empty padding columns.
+           */
+          if (dataIndex === null) { return; }
+
+          const index = parseInt(dataIndex, 10);
+          const value = this.items[index];
+
+          this.ionChange.emit(value);
+        }, 250);
+      })
+    };
+
+    /**
+     * Wrap this in an raf so that the scroll callback
+     * does not fire when component is initially shown.
+     */
+    raf(() => {
+      el.addEventListener('scroll', scrollCallback);
+
+      this.destroyScrollListener = () => {
+        el.removeEventListener('scroll', scrollCallback);
+      }
+    });
+  }
+
+  get activeItem() {
+    return getElementRoot(this.el).querySelector(`.picker-item[data-value="${this.value}"]`) as HTMLElement | null;
```

Comment:
Picker column should be updating `this.value` internally (otherwise we run into the issue you mentioned in https://github.com/ionic-team/ionic-framework/pull/23995#discussion_r719710817). It was previously not doing that.

I am assuming this is no longer relevant now that we are updating `this.value` internally, but let me know if I assumed incorrectly!